Magento_Catalog/templates/product/list/items.phtml

```diff
@@ -0,0 +1,308 @@
+            $description = false;
+            $canItemsAddToCart = false;
+        }
+    break;
+
+    case 'crosssell-rule':
+        /** @var \Magento\Catalog\Block\Product\ProductList\Crosssell $block */
+        if ($exist = $block->hasItems()) {
+            $type = 'crosssell';
+            $class = $type;
+
+            $image = 'cart_cross_sell_products';
+            $title = __('More Choices:');
+            $items = $block->getItemCollection();
+
+            $showAddTo = true;
+            $showCart = true;
+            $templateType = \Magento\Catalog\Block\Product\ReviewRendererInterface::SHORT_VIEW;
+            $description = false;
+            $canItemsAddToCart = false;
+        }
+    break;
+
+    case 'crosssell':
+        /** @var \Magento\Catalog\Block\Product\ProductList\Crosssell $block */
+        if ($exist = count($block->getItems())) {
+            $type = 'crosssell';
+            $class = $type;
+
+            $image = 'cart_cross_sell_products';
+            $title = __('More Choices:');
+            $items = $block->getItems();
+
+            $showAddTo = true;
+            $showCart = true;
+            $templateType = \Magento\Catalog\Block\Product\ReviewRendererInterface::SHORT_VIEW;
+            $description = false;
+            $canItemsAddToCart = false;
+        }
+    break;
+
+    case 'new':
+        if ($exist = $block->getProductCollection()) {
+            $type = 'new';
+            $mode = 'grid';
+            $type = $type . ' ' . $mode;
+
+            $class = 'widget' . ' ' . $type;
+
+            $image = 'new_products_content_widget_grid';
+            $title = __('New Products');
+            $items = $exist;
+
+            $showAddTo = true;
+            $showCart = true;
+            $templateType = \Magento\Catalog\Block\Product\ReviewRendererInterface::SHORT_VIEW;
+            $description = ($mode == 'list') ? true : false;
+            $canItemsAddToCart = false;
+        }
+    break;
+
+    case 'other':
+    break;
+}
+?>
+
+<?php if ($exist): ?>
+    <?php if ($type == 'related' || $type == 'upsell'): ?>
+        <?php if ($type == 'related'): ?>
+            <div
+                class="product-view__slider <?= /* @escapeNotVerified */ $class ?>"
+                data-limit="<?= /* @escapeNotVerified */ $limit ?>"
+                data-shuffle="<?= /* @escapeNotVerified */ $shuffle ?>"
+            >
+        <?php else: ?>
+            <div
+                class="product-view__slider <?= /* @escapeNotVerified */ $class ?>"
+                data-mage-init='{ "upsellProducts": {} }'
+                data-limit="<?= /* @escapeNotVerified */ $limit ?>"
+                data-shuffle="<?= /* @escapeNotVerified */ $shuffle ?>"
+            >
+        <?php endif; ?>
+    <?php else: ?>
+        <div class="product-view__slider <?= /* @escapeNotVerified */ $class ?>">
+    <?php endif; ?>
+
+    <h4
+        id="block-<?= /* @escapeNotVerified */ $class ?>-heading"
```

Comment:
you can leave it in the same line. It's shorter than 80 chars.

Reply:
o, that's true. It must have more unnecessary parameters in the past or just I do it automatically 😄